cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -125,6 +125,16 @@ public NamespaceMeta load(NamespaceId namespaceId) throws Exception {
         return fetchNamespaceMeta(namespaceId);
       }
     });
+    String masterPrincipal = cConf.get(Constants.Security.CFG_CDAP_MASTER_KRB_PRINCIPAL);
+    try {
+      if (cConf.getBoolean(Constants.Security.ENABLED) && SecurityUtil.isKerberosEnabled(cConf)) {
```

Comment:
@anwar6953 I remember we have similar/same logic in some class. Is there a common place for this?